modules: Update trusted-firmware-m for picolibc

keith-packard opened the pull request:
This adds linker script bits and compiler options so that trusted-firmware-m will build with picolibc.

This has not been merged to the Zephyr trusted-firmware-m repository yet, that PR is https://github.com/zephyrproject-rtos/trusted-firmware-m/pull/134

keith-packard commented:
This is waiting for zephyr SDK 0.17.4

nandojve commented:
Hi @keith-packard ,

I would like to printout that #94470 shows that arm platform needs some fixes. I'm not sure if that affects your CI tests.

keith-packard commented:
> Hi @keith-packard ,
> 
> I would like to printout that #94470 shows that arm platform needs some fixes. I'm not sure if that affects your CI tests.

Thanks. It's also blocked on the SDK at present, so once that's merged I'll see what else is broken.

keith-packard commented:
> > Hi @keith-packard ,
> > I would like to printout that #94470 shows that arm platform needs some fixes. I'm not sure if that affects your CI tests.
> 
> Thanks. It's also blocked on the SDK at present, so once that's merged I'll see what else is broken.

All seems to work fine with just this patch.

tomi-font reviewed (dismissed):
just need some competent people (not me) to review the TF-M changes

keith-packard committed:
modules: Update trusted-firmware-m for picolibc

This adds linker script bits and compiler options so that
trusted-firmware-m will build with picolibc.

This has not been merged to the Zephyr trusted-firmware-m repository yet,
that PR is:

https://github.com/zephyrproject-rtos/trusted-firmware-m/pull/134

Signed-off-by: Keith Packard <keithp@keithp.com>

keith-packard committed:
modules: Update tf-m-tests for picolibc

This module also needs some small changes to build with picolibc.

This has not been merged to the Zephyr tf-m-tests repository yet,
that PR is:

https://github.com/zephyrproject-rtos/tf-m-tests/pull/17

Signed-off-by: Keith Packard <keithp@keithp.com>

keith-packard commented:
re-opening without the tf-m-tests change

tomi-font commented:
Probably because you have force-pushed your branch since the close?
